contracts/interfaces/draft-IERC7786.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Interface for ERC-7786 source gateways.
+ *
+ * See ERC-7786 for more details
+ */
+interface IERC7786GatewaySource {
+    /**
+     * @dev Event emitted when a message is created. If `outboxId` is zero, no further processing is necessary, and
+     * not {MessageSent} event SHOULD be expected. If `outboxId` is not zero, then further (gateway specific, and non
+     * standardized) action is required.
+     */
+    event MessageCreated(
+        bytes32 outboxId,
+        string sender, // CAIP-10 account ID
+        string receiver, // CAIP-10 account ID
+        bytes payload,
+        bytes[] attributes
+    );
+
+    /**
+     * @dev This event is emitted when a message, for which the {MessageCreated} event contains an non zero `outboxId`,
+     * received the required post processing actions, and was thus sent to the destination chain.
+     */
+    event MessageSent(bytes32 indexed outboxId);
+
+    /// @dev This error is thrown when a message creation fails because of an unsupported attribute being specified.
+    error UnsuportedAttribute(bytes4 selector);
+
+    /// @dev Getter to check weither an attribute is supported.
+    function supportsAttribute(bytes4 selector) external view returns (bool);
```

Comment:
```suggestion
    /// @dev Getter to check whether an attribute is supported.
```

Reply:
not sure what the suggestion is here.

Reply:
Seems like something got borked with the VSCode plugin. Will fix sorry.